client/client.go

```diff
@@ -148,7 +148,8 @@ func (c *Client) DoRequest(method string, requestURL url.URL, body interface{})
 	if err != nil {
 		return nil, err
 	}
-	return utils.ParseHTTPStatusCodeInResponse(response)
+
+	return response, nil
```

Comment:
Non-2XX status codes can't always be considered failures, so now we return the response and a nil error.